### ISSUE TITLE
"Sign in with SkyID" should use the same portal as the Skapp

### ISSUE THREAD

**Issue by Delivator** [DaWe35/SkyID#3]
It looks like as of right now the `Sign in with SkyID` popup is hard coded to siasky.net.
Imo that popup should use the same portal as the Skapp requesting a sign in when possible.

![msedge_v1TqOynkVe](https://user-images.githubusercontent.com/11231245/97857445-0d93e800-1cfe-11eb-859a-b7d4ddecd543.png)

**Comment by mike76-dev** [comment]
SkyID uses siasky.net in DevMode by default; when not in DevMode, default skynet-js portal is used (which is again siasky.net).
Should allow specifying custom portal in both cases.

**Comment by DaWe35** [comment]
@mike76-dev I think Delivator mentioned that the SkyID connect page always opens on siasky.net: https://github.com/DaWe35/SkyID/blob/95d372a9689c3368126b8520e58f0f979ad012a2/src/skyid.js#L89
I don't know how we can solve this btw.

**Comment by mike76-dev** [comment]
Can't you just add Portal option to this.opts? Then you can pass it on to `new SkynetClient()`.
Alternatively, you can use a global skynetPortal variable and access it via setDefaultPortal() or something like that.

**Comment by DaWe35** [comment]
> 
> 
> Can't you just add Portal option to this.opts? Then you can pass it on to `new SkynetClient()`.
> Alternatively, you can use a global skynetPortal variable and access it via setDefaultPortal() or something like that.

Yeah but the thing is that we want to open SkyID on the portal that you're currently using. So if you open https://sky-note.hns.skyportal.xyz we should open the popup with skyportal.xyz url

**Comment by mike76-dev** [comment]
So, we can split the problem into two parts:
1) Find out the portal where the page is open
2) Pass this portal as an option to the popup
Can you do No.1?

**Comment by DaWe35** [comment]
@mike76-dev When I started write this comment I thought it is impossoble, because there are portals running on subdomains, for example https://skynet.tutemwesi.com. But then I realized we already have a code what does something similar. I'll check it out.

**Comment by DaWe35** [comment]
Redsolver's solution: `final portal = host.split('.hns.').last;`

**Comment by mike76-dev** [comment]
That should work.
But what about the case when the url is entered as `https://siasky.net/hns/domain`?

**Comment by DaWe35** [comment]
I think SkyID already redirects it to domain.hns.siasky.net. I will double-check, thank you!
